src/v2i-hub/ERVCloudForwardingPlugin/src/ERVCloudForwardingPlugin.cpp

```diff
@@ -0,0 +1,214 @@
+    void ERVCloudForwardingPlugin::UpdateConfigSettings()
+    {
+        std::lock_guard<mutex> lock(_cfgLock);
+        GetConfigValue<string>("WebServiceIP", _webIp);
+        GetConfigValue<uint16_t>("WebServicePort", _webPort);
+        GetConfigValue<string>("RSUIp", _rsuIp);
+        GetConfigValue<uint16_t>("SNMPPort", _snmpPort);
+        GetConfigValue<string>("SecurityUser", _securityUser);
+        GetConfigValue<string>("AuthPassPhrase", _authPassPhrase);
+        GetConfigValue<string>("GPSOID", _GPSOID);
+        GetConfigValue<string>("RSUName", _rsuName);
+    }
+
+    void ERVCloudForwardingPlugin::BroadcastBSM(const string &bsmHex)
+    {
+        std::unique_ptr<BsmEncodedMessage> msg;
+        J2735MessageFactory factory;
+        msg.reset();
+        msg.reset(dynamic_cast<BsmEncodedMessage *>(factory.NewMessage(api::MSGSUBTYPE_BASICSAFETYMESSAGE_STRING)));
+        msg->set_payload(bsmHex);
+        msg->set_flags(IvpMsgFlags_RouteDSRC);
+        msg->addDsrcMetadata(0x20);
+        msg->set_encoding(_HEXENC);
+        msg->refresh_timestamp();
+        auto rMsg = dynamic_cast<routeable_message *>(msg.get());
+        BroadcastMessage(*rMsg);
+        PLOG(logDEBUG) << "Broadcast ERV BSM:" << msg->get_payload() << endl;
+    }
+
+    void ERVCloudForwardingPlugin::CARMACloudResponseHandler(QHttpEngine::Socket *socket)
+    {
+        QString st;
+        while (socket->bytesAvailable() > 0)
+        {
+            auto readBytes = socket->readAll();
+            st.append(readBytes);
+        }
+        QByteArray array = st.toLocal8Bit();
+        const char *_cloudUpdate = array.data();
+        string bsmHex = _cloudUpdate;
+        PLOG(logINFO) << "Received ERV BSM from cloud:" << bsmHex << endl;
+        BroadcastBSM(bsmHex);
+    }
+
+    int ERVCloudForwardingPlugin::StartBSMWebService()
+    {
+        char *placeholderX[1] = {nullptr};
+        int placeholderC = 1;
+        QCoreApplication a(placeholderC, placeholderX);
+        auto address = QHostAddress(QString::fromStdString(_webIp));
+        auto port = static_cast<quint16>(_webPort);
+        QSharedPointer<OpenAPI::OAIApiRequestHandler> handler(new OpenAPI::OAIApiRequestHandler());
+        handler = QSharedPointer<OpenAPI::OAIApiRequestHandler>(new OpenAPI::OAIApiRequestHandler());
+        auto router = QSharedPointer<OpenAPI::OAIApiRouter>::create();
+        router->setUpRoutes();
+        QObject::connect(handler.data(), &OpenAPI::OAIApiRequestHandler::requestReceived, [&](QHttpEngine::Socket *socket)
+                         { CARMACloudResponseHandler(socket); });
+        QObject::connect(handler.data(), &OpenAPI::OAIApiRequestHandler::requestReceived, [&](QHttpEngine::Socket *socket)
+                         { router->processRequest(socket); });
+        QHttpEngine::Server server(handler.data());
+        if (!server.listen(address, port))
+        {
+            qCritical("Unable to listen on the specified port.");
+            return 1;
```

Comment:
Is there a reason why we return an int for this?

Reply:
I think 1 represent failure.